src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -59,63 +51,88 @@ final class SettingsTable extends AbstractVirtualTable
+                                     AbstractMutableVirtualTable.ColumnValues clusteringColumns,
+                                     Optional<AbstractMutableVirtualTable.ColumnValue> columnValue)
+    {
+        String name = partitionKey.value(0);
+        String value = columnValue.map(v -> v.value().toString()).orElse(null);
+        runExceptionally(() -> DatabaseDescriptor.setProperty(getKeyAndWarn(name), value),
+                         t -> invalidRequest("Invalid update request for property '%s'. %s", name, t.getMessage()));
     }
 
     @Override
     public DataSet data(DecoratedKey partitionKey)
     {
         SimpleDataSet result = new SimpleDataSet(metadata());
         String name = UTF8Type.instance.compose(partitionKey.getKey());
-        if (BACKWARDS_COMPATABLE_NAMES.containsKey(name))
-            ClientWarn.instance.warn("key '" + name + "' is deprecated; should switch to '" + BACKWARDS_COMPATABLE_NAMES.get(name) + "'");
-        if (PROPERTIES.containsKey(name))
-            result.row(name).column(VALUE, getValue(PROPERTIES.get(name)));
+        try
+        {
+            Object value = getStringProperty(getKeyAndWarn(name));
+            result.row(name).column(VALUE, value);
+        }
+        catch (PropertyNotFoundException e)
+        {
+            // ignore
+        }
         return result;
     }
 
     @Override
     public DataSet data()
     {
         SimpleDataSet result = new SimpleDataSet(metadata());
-        for (Map.Entry<String, Property> e : PROPERTIES.entrySet())
-            result.row(e.getKey()).column(VALUE, getValue(e.getValue()));
+        DatabaseDescriptor.visit(withBackwardsCompatableNamesVisitor((key, type, ro) ->
+                                                                result.row(key).column(VALUE, getStringProperty(BACKWARDS_COMPATABLE_NAMES.getOrDefault(key, key)))),
+                                  t -> new ConfigurationException(t.getMessage(), false));
         return result;
     }
 
-    private String getValue(Property prop)
+    static String getKeyAndWarn(String name)
     {
-        Object value = prop.get(config);
-        return value == null ? null : value.toString();
+        String key = BACKWARDS_COMPATABLE_NAMES.getOrDefault(name, name);
+        if (BACKWARDS_COMPATABLE_NAMES.containsKey(name))
+            ClientWarn.instance.warn("key '" + name + "' is deprecated; should switch to '" + BACKWARDS_COMPATABLE_NAMES.get(name) + '\'');
+        return key;
+    }
+
+    static DatabaseDescriptor.PropertyVisitor withBackwardsCompatableNamesVisitor(DatabaseDescriptor.PropertyVisitor visitor)
+    {
+        BiMap<String, String> inversedNames = BACKWARDS_COMPATABLE_NAMES.inverse();
+        return (name, type, readOnly) -> {
+            if (inversedNames.containsKey(name))
+                visitor.visit(inversedNames.get(name), type, readOnly);
+            visitor.visit(name, type, readOnly);
+        };
```

Comment:
Can you remove this concept, this adds more complexity than we need... We need a few things

# On Init

8 did someone add a new config that conflicts with one of the old names defined by this class?  If so fail fast (needed so that config doesn't get added and is blocked before merge)

# At runtime

* any usage of the old name gets mapped to the new name

given this, `getKeyAndWarn(name)` is all we need, and a `visitor` is too heavy weight.

Reply:
You're right. This is unnecessary, so I have removed it.